packages/core/test/actorLogic.test.ts

```diff
@@ -464,6 +465,44 @@ describe('promise logic (fromPromise)', () => {
     const fn2 = signalListenerList[1];
     expect(fn2).toHaveBeenCalled();
   });
+
+  it('can emit updates', async () => {
+    const p = fromPromise(async ({ update }) => {
+      update({ value: 'starting' });
+      update({ value: 'loading', context: { progress: 0.6 } });
+      update({ value: 'finished' });
+      return 'done data';
+    });
+
+    const stuff: Array<{}> = [];
+
+    const actor = createActor(p);
+    actor.subscribe((s) => {
+      stuff.push(s);
+    });
+    actor.start();
+
+    const res = await toPromise(actor);
+
+    expect(res).toEqual('done data');
+
+    expect(stuff).toEqual(
+      expect.arrayContaining([
+        expect.objectContaining({}),
+        expect.objectContaining({ value: 'starting', context: undefined }),
+        expect.objectContaining({
+          value: 'loading',
+          context: { progress: 0.6 }
+        }),
+        expect.objectContaining({ value: 'finished', context: undefined }),
```

Comment:
I feel like this test case is too big. "can emit updates" is too broad and doesn't give me much insight into what is actually expected because it reads as "it works".

For instance, I noticed that an update with just one of those fields would wipe out the other one. I was wondering how deliberate that was and if we have a test for it. It turns out we do - but it's just so buried in this item here. It's hard to tell if this was intentional or accidental. I think it was intentional and that's fine - but my point stands: a test with so many details in it isn't particularly informative/explicit about its intentions